src/main/java/net/dv8tion/jda/api/entities/Message.java

```diff
@@ -1117,6 +1117,13 @@ public Pattern getPattern()
         private final int size;
         private final int height;
         private final int width;
+        
+        /**
+        * Pattern used to check if the file is an image.
+        *
+        * @see #isImage()
+        */
+        private final Pattern image = Pattern.compile("([^\\s]+)(?:\\.(?i)(jpe?g|png|gif|bmp|webp))$");
```

Comment:
Why use a regex for this? This is trivial to implement without a regex and would probably be less error-prone than this.

```java
int index = name.lastIndexOf('.');
if (index + 1 == name.length())
    return false;
String ext = name.substring(index + 1);
return "png|webp|jpeg|jpg|gif".contains(ext);
```

Also I'm pretty sure bmp is not supported.

Reply:
You're right probably. I'll leave the decision up to you and other maintainers.